tiny-event-sourcing-lib/src/main/kotlin/ru/quipy/streams/EventStreamReaderManager.kt

```diff
@@ -32,38 +37,48 @@ class ActiveEventStreamReaderManager(
         return true
     }
 
-    override fun tryInterceptReading(streamName: String): Boolean {
+    override fun tryInterceptReading(streamName: String, readerId: String): Boolean {
         val currentActiveReader: ActiveEventStreamReader? = eventStore.getActiveStreamReader(streamName)
-        val newActiveReader = createNewActiveReader(streamName, currentActiveReader)
+
+        if (currentActiveReader != null && currentActiveReader.readerId == readerId) {
+            logger.info("An attempt to intercept reading of stream $streamName by its active reader $readerId")
+            return false
+        }
+
+        val newActiveReader = createNewActiveReader(streamName, readerId, currentActiveReader)
 
         val expectedVersion = currentActiveReader?.version ?: 0
 
         if (eventStore.tryReplaceActiveStreamReader(expectedVersion, newActiveReader)) {
-            logger.info("Event stream reader of stream $streamName has been switched from [${currentActiveReader?.id}] to [${newActiveReader.id}]")
+            logger.info("Event stream reader of stream $streamName has been switched from [${currentActiveReader?.readerId}] to [${newActiveReader.readerId}]")
             return true
         }
 
         return false
     }
 
-    override fun updateReaderState(streamName: String, readingIndex: Long) {
+    override fun updateReaderState(streamName: String, readerId: String, readingIndex: Long) {
         val activeReader: ActiveEventStreamReader? = eventStore.getActiveStreamReader(streamName)
 
+        if (activeReader != null && activeReader.readerId != readerId)
+            return
+
         val updatedActiveReader = ActiveEventStreamReader(
                 activeReader?.id ?: streamName,
                 activeReader?.version ?: 1,
```

Comment:
`activeReader?.version + 1 ?: 0?`

Reply:
have you forgotten to increase the version?

Reply:
no, you haven't. but it looks strange as the version will not be updated. So many instances may update it concurrently as they wish?

Reply:
The PREVIOUS version should be matched but the version should be updated at the same time